lib/lib-upload/src/main/java/org/sagebionetworks/upload/multipart/MultipartUploadUtils.java

```diff
@@ -0,0 +1,63 @@
+package org.sagebionetworks.upload.multipart;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.sagebionetworks.repo.model.upload.PartRange;
+
+public class MultipartUploadUtils {
+
+	protected static final int MAX_PARTS_IN_ONE_COMPOSE = 32;
+
+	/**
+	 * Create a part key using the base and part number.
+	 *
+	 * @param baseKey
+	 * @param partNumber
+	 * @return
+	 */
+	public static String createPartKey(String baseKey, long partNumber) {
+		return String.format("%1$s/%2$d", baseKey, partNumber);
+	}
+
+	/**
+	 * Create a part key using the base and part range.
+	 *
+	 * @param baseKey
+	 * @param lowerBound
+	 * @param upperBound
+	 * @return
+	 */
+	public static String createPartKeyFromRange(String baseKey, long lowerBound, long upperBound) {
+		if (lowerBound==upperBound) {
+			return createPartKey(baseKey, lowerBound);
+		}
+		return String.format("%1$s/%2$d-%3$d", baseKey, lowerBound, upperBound);
+	}
+
+	public static List<PartRange> getListOfPartRangesToLookFor(long lowerBound, long upperBound, long maxPartNumber) {
+		long currentPartSize = upperBound - lowerBound + 1;
+		currentPartSize = computeCurrentPartSize(currentPartSize);
+		long newPartSize = currentPartSize * MAX_PARTS_IN_ONE_COMPOSE;
+		long newLowerBound = computeNewLowerBound(lowerBound, newPartSize);
+		long newUpperBound = Math.min(newLowerBound + newPartSize - 1, maxPartNumber);
+		List<PartRange> partRanges = new ArrayList<>();
```

Comment:
Don't need these elements. Just need a count. Then check the database if that count exists between that range.

Reply:
Also need to verify the sum of all of the part ranges in [newLowerBound, newUpperBound]